pallets/runtime/extensions/src/check_weight.rs

```diff
@@ -0,0 +1,737 @@
+                    .ok_or(InvalidTransaction::ExhaustsResources)?;
+                all_weight
+                    .checked_add(extrinsic_weight, DispatchClass::Operational)
+                    .map_err(|_| InvalidTransaction::ExhaustsResources)?;
+
+                // If it would fit in normally, its okay
+                if all_weight.total() <= maximum_weight ||
+                    // If we have not used our operational space
+                    all_weight.get(DispatchClass::Operational) <= operational_space
+                {
+                    Ok(all_weight)
+                } else {
+                    Err(InvalidTransaction::ExhaustsResources.into())
+                }
+            }
+        }
+    }
+
+    /// Checks if the current extrinsic can fit into the block with respect to block length limits.
+    ///
+    /// Upon successes, it returns the new block length as a `Result`.
+    fn check_block_length(
+        info: &DispatchInfoOf<T::Call>,
+        len: usize,
+    ) -> Result<u32, TransactionValidityError> {
+        let current_len = Module::<T>::all_extrinsics_len();
+        let maximum_len = T::MaximumBlockLength::get();
+        let limit = Self::get_dispatch_limit_ratio(info.class) * maximum_len;
+        let added_len = len as u32;
+        let next_len = current_len.saturating_add(added_len);
+        if next_len > limit {
+            Err(InvalidTransaction::ExhaustsResources.into())
+        } else {
+            Ok(next_len)
+        }
+    }
+
+    /// Creates new `SignedExtension` to check weight of the extrinsic.
+    pub fn new() -> Self {
+        Self(Default::default())
+    }
+
+    /// Do the pre-dispatch checks. This can be applied to both signed and unsigned.
+    ///
+    /// It checks and notes the new weight and length.
+    fn do_pre_dispatch(
+        info: &DispatchInfoOf<T::Call>,
+        len: usize,
+    ) -> Result<(), TransactionValidityError> {
+        let next_len = Self::check_block_length(info, len)?;
+        let next_weight = Self::check_block_weight(info)?;
+        Self::check_extrinsic_weight(info)?;
+
+        AllExtrinsicsLen::put(next_len);
+        BlockWeight::put(next_weight);
+        Ok(())
+    }
+
+    /// Do the validate checks. This can be applied to both signed and unsigned.
+    ///
+    /// It only checks that the block weight and length limit will not exceed.
+    fn do_validate(info: &DispatchInfoOf<T::Call>, len: usize) -> TransactionValidity {
+        // ignore the next length. If they return `Ok`, then it is below the limit.
+        let _ = Self::check_block_length(info, len)?;
+        // during validation we skip block limit check. Since the `validate_transaction`
+        // call runs on an empty block anyway, by this we prevent `on_initialize` weight
+        // consumption from causing false negatives.
+        Self::check_extrinsic_weight(info)?;
+
+        Ok(ValidTransaction::default())
```

Comment:
Do we want to add a doc comment here that we changed the logic here in the favor of Polymesh?

Reply:
There is a small comment on the top about changes of Polymath, and I does not hurt to repeat that here.